base/error.jl

```diff
@@ -23,7 +23,8 @@ error(s...) = throw(Main.Base.ErrorException(Main.Base.string(s...)))
 
 rethrow() = ccall(:jl_rethrow, Void, ())::Bottom
 rethrow(e) = ccall(:jl_rethrow_other, Void, (Any,), e)::Bottom
-backtrace() = ccall(:jl_backtrace_from_here, Array{Ptr{Void},1}, ())
+backtrace() = ccall(:jl_backtrace_from_here, Array{Ptr{Void},1}, (Int32,), false)
+backtrace2() = ccall(:jl_backtrace_from_here, Ref{SimpleVector}, (Int32,), true)
```

Comment:
it wouldn't be a type-stable return, and i don't really want to export this behavior yet. it's mostly just being exposed so that it can be experimented with, and likely developed into a proper api eventually (some sort of semi-complete wrapper around libunwind-like functionality, probably).

that said, i can take out this stub if you would prefer and maintain this portion as a local patch. i've been holding off merging until i get some thoughts on that.